workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/GameObjectRepresentation/SpatialOSBehaviourLibraryTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.CodeDom;
+using System.Linq;
+using System.Text.RegularExpressions;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Object = UnityEngine.Object;
+
+namespace Improbable.Gdk.Core.EditmodeTests
+{
+    [TestFixture]
+    public class SpatialOSBehaviourLibraryTests
+    {
+        private const uint ComponentId1 = 1337;
+        private const uint ComponentId2 = 1338;
+
+        [ReaderInterface]
+        [ComponentId(ComponentId1)]
+        private interface DummyReader
+        {
+        }
+
+        [WriterInterface]
+        [ComponentId(ComponentId1)]
+        private interface DummyWriter1
+        {
+        }
+
+        [WriterInterface]
+        [ComponentId(ComponentId2)]
+        private interface DummyWriter2
+        {
+        }
+
+        private class SingleReaderBehaviour : MonoBehaviour
+        {
+            [Require] public DummyReader Reader;
+        }
+
+        private class TwoWritersBehaviour : MonoBehaviour
+        {
+            [Require] public DummyWriter1 Writer1;
+            [Require] public DummyWriter2 Writer2;
+        }
+
+        private class MultipleReadersOfSameType : MonoBehaviour
+        {
+            [Require] public DummyReader Reader1;
+            [Require] public DummyReader Reader2;
+        }
+
+        private class RequireInvalidMember : MonoBehaviour
+        {
+            [Require] public int Bad;
+        }
+
+        private SpatialOSBehaviourLibrary library;
+        private GameObject testGameObject;
+
+        [SetUp]
+        public void Setup()
+        {
+            library = new SpatialOSBehaviourLibrary(new LoggingDispatcher());
+            testGameObject = new GameObject();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            if (testGameObject != null)
+            {
+                Object.DestroyImmediate(testGameObject);
+            }
+        }
+
+        [Test]
+        public void SpatialOSBehaviourLibrary_injects_Reader()
+        {
+            // TODO - Needs ReaderWriterFactory implemented first!
+            Assert.Pass();
+        }
+
+        [Test]
+        public void SpatialOSBehaviourLibrary_injects_two_Writers()
+        {
+            // TODO - Needs ReaderWriterFactory implemented first!
```

Comment:
How about asserting that it throws `NotImplementedException` so tests can't be forgotten?

Reply:
Good point now that we are merging to a feature branch and don't need all tests to pass.